src/Business Foundation/App/NoSeries/src/Single/NoSeriesStatelessImpl.Codeunit.al

```diff
@@ -0,0 +1,119 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Foundation.NoSeries;
+
+codeunit 306 "No. Series - Stateless Impl." implements "No. Series - Single"
+{
+    Access = Internal;
+    Permissions = tabledata "No. Series Line" = rimd,
+                  tabledata "No. Series" = r;
+
+    var
+        NumberLengthErr: Label 'The number %1 cannot be extended to more than 20 characters.', comment = '%1=No.';
+
+    procedure PeekNextNo(NoSeriesLine: Record "No. Series Line"; UsageDate: Date): Code[20]
+    begin
+        exit(GetNextNoInternal(NoSeriesLine, false, UsageDate, false));
+    end;
+
+    procedure GetNextNo(var NoSeriesLine: Record "No. Series Line"; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    begin
+        exit(GetNextNoInternal(NoSeriesLine, true, UsageDate, HideErrorsAndWarnings));
+    end;
+
+    local procedure GetNextNoInternal(var NoSeriesLine: Record "No. Series Line"; ModifySeries: Boolean; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        NoSeriesMgtInternal: Codeunit NoSeriesMgtInternal;
+    begin
+        if UsageDate = 0D then
+            UsageDate := WorkDate();
+        if NoSeriesLine."Last No. Used" = '' then begin
+            if HideErrorsAndWarnings and (NoSeriesLine."Starting No." = '') then
+                exit('');
+            NoSeriesLine.TestField("Starting No.");
+            NoSeriesLine."Last No. Used" := NoSeriesLine."Starting No.";
+        end else
+            if NoSeriesLine."Increment-by No." <= 1 then
+                NoSeriesLine."Last No. Used" := IncStr(NoSeriesLine."Last No. Used")
+            else
+                IncrementNoText(NoSeriesLine."Last No. Used", NoSeriesLine."Increment-by No.");
+
+        if not NoSeriesMgtInternal.EnsureLastNoUsedIsWithinValidRange(NoSeriesLine, HideErrorsAndWarnings) then
+            exit('');
```

Comment:
In general we are trying to be explicit in what is being returned to avoid any confusion, even if it's the default value to make it more maintainable.